### PR TITLE
Upgrade to Rails 7.1

### LINKED ISSUE #1062: Upgrade to rails 7.1

### PR DESCRIPTION
**Pull request by thatbudakguy**
Closes #1062

Spun out from #963

### CODE REVIEW

File: config/initializers/new_framework_defaults_6_1.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@
 # Use new connection handling API. For most applications this won't have any
 # effect. For applications using multiple databases, this new API provides
 # support for granular connection swapping.
-Rails.application.config.active_record.legacy_connection_handling = false
+# Rails.application.config.active_record.legacy_connection_handling = false
```
Comment (jcoyne) [inline config/initializers/new_framework_defaults_6_1.rb:45]: Was this change necessary?

**Review by jcoyne — APPROVED** [review]
One question for consideration, then feel free to merge.

File: config/initializers/new_framework_defaults_6_1.rb (hunk 1 above)
Comment (thatbudakguy) [reply config/initializers/new_framework_defaults_6_1.rb:45]: Dunno, I think it was automatically added like everything else? I can leave it out if desired. No idea what the impact is
